daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -579,13 +579,15 @@ object Engine {
         List.empty[Versioned[DisclosedContract]],
       )
     ) { case ((m1, m2, ds), d) =>
+      // TODO (drsk) drop this conversion from SValue -> Value. 
+      // https://github.com/digital-asset/daml/issues/14069
       val arg = machine.normValue(d.templateId, d.argument)
       val coid = machine.normValue(d.templateId, d.contractId)
       val version = machine.tmplId2TxVersion(d.templateId)
 
       // check for well typed contract argument
       (coid, arg) match {
-        case (ValueContractId(coid), r @ ValueRecord(Some(d.templateId), _)) =>
+        case (ValueContractId(coid), r @ ValueRecord(None, _)) =>
```

Comment:
```suggestion
        case (ValueContractId(coid), r) =>
```
don’t really see a reason to match on this stuff. If you need the record info (haven’t checked) match on `ValueRecord(_, _)`.